server.js

```diff
@@ -1,9 +1,15 @@
 const express = require("express");
 const path = require("path");
+const jsonServer = require("json-server");
+const router = jsonServer.router("db.json");
+const middlewares = jsonServer.defaults();
 
 const app = express();
 const port = process.env.PORT || 3001;
 
+app.use(middlewares);
+app.use("/api", router);
+
```

Comment:
Your Heroku deployment will not work that way. You better move this lines of code above the `app.get("*", (req... ` part.